restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/SmsInterpreter.java

```diff
@@ -378,7 +388,24 @@ public void onReceive(final Object message) throws Exception {
             if (Verbs.redirect.equals(verb.name())) {
                 fsm.transition(message, redirecting);
             } else if (Verbs.sms.equals(verb.name())) {
-                fsm.transition(message, creatingSmsSession);
+                //Check if Outbound SMS is allowed
+                ExtensionController ec = ExtensionController.getInstance();
+                final IExtensionFeatureAccessRequest far = new FeatureAccessRequest(FeatureAccessRequest.Feature.OUTBOUND_SMS, accountId);
+                ExtensionResponse er = ec.executePreInboundAction(far, this.extensions);
+
+                if (er.isAllowed()) {
+                    fsm.transition(message, creatingSmsSession);
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        final String errMsg = "Outbound SMS is not Allowed";
+                        logger.debug(errMsg);
+                    }
+                    final NotificationsDao notifications = storage.getNotificationsDao();
+                    final Notification notification = notification(WARNING_NOTIFICATION, 11001, "Outbound SMS is now allowed");
+                    notifications.addNotification(notification);
+                    fsm.transition(message, finished);
```

Comment:
are we covering postaction invocation later as part of FSM?

Reply:
@jaimecasero no, I missed the `post` action and I will add it now